### ISSUE TITLE
Unable to restore a Gitea dump

### ISSUE THREAD

**Issue by mrv96** [go-gitea/gitea#15678]
- Gitea version (or commit ref): 1.14.1
- Git version: 2.25.1
- Operating system: Ubuntu 20.04
  
  
  
- Database:
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] NA
- Log gist:

```
2021/04/30 16:24:26 cmd/web.go:82:runWeb() [I] Starting Gitea on PID: 9060
2021/04/30 16:24:26 ...dules/setting/git.go:101:newGit() [I] Git Version: 2.25.1, Wire Protocol Version 2 Enabled
2021/04/30 16:24:26 cmd/web.go:126:runWeb() [I] Global init
2021/04/30 16:24:26 ...dules/setting/git.go:101:newGit() [I] Git Version: 2.25.1, Wire Protocol Version 2 Enabled
2021/04/30 16:24:26 routers/init.go:134:GlobalInit() [T] AppPath: /usr/local/bin/gitea
2021/04/30 16:24:26 routers/init.go:135:GlobalInit() [T] AppWorkPath: /var/lib/gitea/
2021/04/30 16:24:26 routers/init.go:136:GlobalInit() [T] Custom path: /var/lib/gitea/custom
2021/04/30 16:24:26 routers/init.go:137:GlobalInit() [T] Log path: /var/lib/gitea/log
2021/04/30 16:24:26 routers/init.go:49:checkRunMode() [I] Run Mode: Prod
2021/04/30 16:24:26 ...dules/setting/log.go:287:newLogService() [I] Gitea v1.14.1 built with GNU Make 4.1, go1.16.3 : bindata, sqlite, sqlite_unlock_notify
2021/04/30 16:24:26 ...dules/setting/log.go:333:newLogService() [I] Gitea Log Mode: Console(Console:info)
2021/04/30 16:24:26 ...dules/setting/log.go:250:generateNamedLogger() [I] Router Log: Console(console:info)
2021/04/30 16:24:26 ...les/setting/cache.go:73:newCacheService() [I] Cache Service Enabled
2021/04/30 16:24:26 ...les/setting/cache.go:88:newCacheService() [I] Last Commit Cache Service Enabled
2021/04/30 16:24:26 ...s/setting/session.go:77:newSessionService() [I] Session Service Enabled
2021/04/30 16:24:26 ...s/storage/storage.go:158:initAttachments() [I] Initialising Attachment storage with type: 
2021/04/30 16:24:26 ...les/storage/local.go:47:NewLocalStorage() [I] Creating new Local Storage at /var/lib/gitea/data/attachments
2021/04/30 16:24:26 ...s/storage/storage.go:152:initAvatars() [I] Initialising Avatar storage with type: 
2021/04/30 16:24:26 ...les/storage/local.go:47:NewLocalStorage() [I] Creating new Local Storage at /var/lib/gitea/data/avatars
2021/04/30 16:24:26 ...s/storage/storage.go:170:initRepoAvatars() [I] Initialising Repository Avatar storage with type: 
2021/04/30 16:24:26 ...les/storage/local.go:47:NewLocalStorage() [I] Creating new Local Storage at /var/lib/gitea/data/repo-avatars
2021/04/30 16:24:26 ...s/storage/storage.go:164:initLFS() [I] Initialising LFS storage with type: 
2021/04/30 16:24:26 ...les/storage/local.go:47:NewLocalStorage() [I] Creating new Local Storage at /var/lib/gitea/data/lfs
2021/04/30 16:24:26 routers/init.go:150:GlobalInit() [I] SQLite3 Supported
2021/04/30 16:24:26 routers/init.go:68:initDBEngine() [I] Beginning ORM engine initialization.
2021/04/30 16:24:26 routers/init.go:75:initDBEngine() [I] ORM engine initialization attempt #1/10...
2021/04/30 16:24:26 ...om/urfave/cli/app.go:524:HandleAction() [I] PING DATABASE mysql
2021/04/30 16:24:26 ...ations/migrations.go:378:Migrate() [F] Gitea no longer supports auto-migration from your previously installed version.
	Please try upgrading to a lower version first (suggested v1.6.4), then upgrade to this version.
```

## Description

Hi, I installed Gitea 1.13.0 and I regularly updated Gitea up to version 1.14.1. Now I'm trying to restore a dump done with version 1.14.1 but I obtain always the auto-migration error visible in the log. I performed several attempts (e.g. Docker, Ubuntu package, binary installation) but everything fails and ends in the already mentioned error.

In the restore process I followed the steps suggested here: https://docs.gitea.io/en-us/backup-and-restore/

At this point I suspects that this is a Gitea issue in some way.

**Comment by zeripath** [comment]
What is the contents of the version table in the dump?

**Comment by mrv96** [comment]
I don't know. How can I check it? My dump contains only app.ini, repo directory and .sql file.

**Comment by zeripath** [comment]
The SQL file will contain SQL commands one of those is going to be creating the version table and inserting the version value. 

The other place is to look in the db itself and check what the contents of the version table is.

**Comment by mrv96** [comment]
Are you talking about this?
```
CREATE TABLE IF NOT EXISTS `version` (`id` BIGINT(20) PRIMARY KEY AUTO_INCREMENT NOT NULL) DEFAULT CHARSET utf8mb4 ROW_FORMAT=DYNAMIC;
INSERT INTO `version` (`id`) VALUES (1);
```
If yes it seems that there is no stored version.

If I query my running Gitea binary installation the result is:
```
mysql> SELECT * FROM gitea.version;
+----+---------+
| id | version |
+----+---------+
|  1 |     178 |
+----+---------+
1 row in set (0.00 sec)
```

**Comment by zeripath** [comment]
Yup tha be the problem.

You need to set that number to whatever it was when you dumped gitea. From 1.14 onwards we now dump the value obviously your dump came before that.

So what version did you dump from?

**Comment by zeripath** [comment]
For 1.13.x that value should be 156

**Comment by mrv96** [comment]
Thank you very much. Your fix works!

I dumped from version 178, and with this two lines
``` mysql
CREATE TABLE IF NOT EXISTS `version` (`id` BIGINT(20) PRIMARY KEY AUTO_INCREMENT NOT NULL, `version` BIGINT(20) NOT NULL) DEFAULT CHARSET utf8mb4 ROW_FORMAT=DYNAMIC;
INSERT INTO `version` (`id`, `version`) VALUES (1, 178);
```
at the end of gitea-db.sql, everything worked.

Hence the problem is that the command `gitea dump` doesn't include the field version in the gitea-db.sql file.
I performed multiple dump attempts, but the `dump` command never creates the `version` field.

Are you able to reproduce the issue?
Is it possible that this issue is due to the multiple gitea versions update (from 1.13.0 to 1.14.1)?
If yes the issue could be in the updating procedure.
